core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -254,7 +254,7 @@ package object config {
       "consumption must fit within some hard limit then be sure to shrink your JVM heap size " +
       "accordingly. This must be set to a positive value when spark.memory.offHeap.enabled=true.")
     .bytesConf(ByteUnit.BYTE)
-    .checkValue(_ >= 0, "The off-heap memory size must not be negative")
+    .checkValue(_ > 0, "The off-heap memory size must be positive.")
```

Comment:
Hi, @xingyif . Thank you for making your first PR. BTW, the PR title seems to be misleading. `Remove value check of MEMORY_OFFHEAP_SIZE` means removing the whole `.checkValue(_ >= 0)` part. Could you update it?

Reply:
If the default is 0, won't this fail? I think you can leave it alone.

Reply:
@dongjoon-hyun Good catch! Changed.

Reply:
@xingyif . @srowen means this change will conflict with line 258 (`.createWithDefault(0)`).

Reply:
Thomas Graves and I had a discussion about this in the [jira](https://issues.apache.org/jira/browse/SPARK-28859) ticket. When user specifies the value, it requires the specified value to be > 0. However, if the user doesn't change the value (default), the it is ok to have it to be 0.


> Default of 0 is fine that is why I said if the user specifies a value it should be > 0